### PR TITLE
fix: improve make it real action

### PR DESCRIPTION
**Pull request by fundon**

### CODE REVIEW

**Comment by fundon** [comment]
* **#6830** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6830?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/ai/provider.tsx [hunk 1]
```diff
@@ -246,12 +246,16 @@ export function setupAIProvider() {
   });
 
   AIProvider.provide('makeItReal', options => {
+    let content = options.content || '';
+    if (options.attachments?.length) {
+      content = `Here are the latest wireframes. Could you make a new website based on these wireframes and notes and send back just the html file?'
+      ${content}
```
Comment (pengx17) [inline packages/frontend/core/src/components/blocksuite/block-suite-editor/ai/provider.tsx]: Remove `'` at the end of line 251
I think the user content may be decorated with some hint like `hint: ${content}`

File: packages/backend/server/src/plugins/copilot/session.ts [hunk 2]
```diff
@@ -143,7 +143,12 @@ export class ChatSession implements AsyncDisposable {
         Object.keys(params).length ? params : firstMessage?.params || {},
         this.config.sessionId
       ),
-      ...messages.filter(m => m.content?.trim() || m.attachments?.length),
+      ...messages
+        .map(m => ({
+          ...m,
+          content: m.content.trim(),
+        }))
+        .filter(m => m.content.length || m.attachments?.length),
```
Comment (pengx17) [inline packages/backend/server/src/plugins/copilot/session.ts]: I think non-chat message use the line 123 to 139. What you modified here is for chat messages.

File: packages/backend/server/src/plugins/copilot/controller.ts [hunk 3]
```diff
@@ -190,6 +190,7 @@ export class CopilotController {
       const session = await this.appendSessionMessage(sessionId, messageId);
       delete params.messageId;
 
+      this.logger.debug(session.finish(params));
```
Comment (darkskygit) [inline packages/backend/server/src/plugins/copilot/controller.ts]: cleanup logging

**Review by pengx17 — APPROVED** [review]

File: packages/backend/server/src/plugins/copilot/providers/openai.ts [hunk 4]
```diff
@@ -80,15 +80,22 @@ export class OpenAIProvider
     // filter redundant fields
     return messages.map(({ role, content, attachments }) => {
       if (Array.isArray(attachments)) {
-        const contents = [
-          { type: 'text', text: content },
-          ...attachments
+        const contents: OpenAI.Chat.Completions.ChatCompletionContentPart[] =
+          [];
+        if (content.length) {
```
Comment (pengx17) [inline packages/backend/server/src/plugins/copilot/providers/openai.ts:86]: .trim before push?

**Review by darkskygit — APPROVED** [review]

**Review by forehalo — APPROVED** [review]

**Pull request merged**